exchanges/bybit/bybit_test.go

```diff
@@ -0,0 +1,3302 @@
+		t.Fatalf("received: '%v' but expected: '%v'", err, common.ErrNilPointer)
+	}
+	var testWg sync.WaitGroup
+	err = b.Start(&testWg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	testWg.Wait()
+}
+
+// test cases for SPOT
+
+func TestGetAllSpotPairs(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetAllSpotPairs(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetOrderBook(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetOrderBook(context.Background(), "BTCUSDT", 100)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetMergedOrderBook(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetMergedOrderBook(context.Background(), "BTCUSDT", 2, 100)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetTrades(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetTrades(context.Background(), "BTCUSDT", 100)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetKlines(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetKlines(context.Background(), "BTCUSDT", "5m", 2000, time.Now().Add(-time.Hour*1), time.Now())
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGet24HrsChange(t *testing.T) {
+	t.Parallel()
+	_, err := b.Get24HrsChange(context.Background(), "BTCUSDT")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = b.Get24HrsChange(context.Background(), "")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetLastTradedPrice(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetLastTradedPrice(context.Background(), "BTCUSDT")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = b.GetLastTradedPrice(context.Background(), "")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetBestBidAskPrice(t *testing.T) {
+	t.Parallel()
+	_, err := b.GetBestBidAskPrice(context.Background(), "BTCUSDT")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = b.GetBestBidAskPrice(context.Background(), "")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestCreatePostOrder(t *testing.T) {
+	t.Parallel()
+
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test: api keys not set or canManipulateRealOrders set to false")
+	}
+	_, err := b.CreatePostOrder(context.Background(), &PlaceOrderRequest{
+		Symbol:      "BTCUSDT",
+		Quantity:    1,
+		Side:        "Buy",
+		TradeType:   "LIMIT",
+		TimeInForce: "GTC",
+		Price:       100,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestQueryOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders { // Note: here !canManipulateRealOrders added as we don't have orderID
+		t.Skip("skipping test: api keys not set or canManipulateRealOrders set to false")
+	}
+	_, err := b.QueryOrder(context.Background(), "0", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestCancelExistingOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test: api keys not set or canManipulateRealOrders set to false")
+	}
+	_, err := b.CancelExistingOrder(context.Background(), "", "")
```

Comment:
You can send in an old order ID and it will return query order data not necessarily cancel an order. So we might need to add in a specific check:
```
if resp.Data.Status == "CANCELED" {
   return nil, fmt.Errorf("%s %w", resp.Data.OrderID, order.ErrAlreadyCancelled)
}
```

Reply:
I didn't get you here. As per my understanding, `resp.Data.Status` will be `CANCELED` when BYBIT is able to successfully process the cancel request. So why do you want to return such error `ErrAlreadyCancelled` from the `CancelExistingOrder` function?

Please share your perspective. @shazbert

Reply:
When canceling the response is below which doesn't change the status state: 
```json
{"ret_code":0,"ret_msg":"","ext_code":null,"ext_info":null,"result":{"accountId":"bruh","symbol":"XRPUSDT","orderLinkId":"1657843037201","orderId":"1200468845536492288","transactTime":"1657843038204","price":"1","origQty":"10","executedQty":"0","status":"NEW","timeInForce":"GTC","type":"LIMIT","side":"SELL"}}
```

When canceling again the response is this with no errors:
```json
{"ret_code":0,"ret_msg":"","ext_code":null,"ext_info":null,"result":{"accountId":"bruh","symbol":"XRPUSDT","orderLinkId":"1657843037201","orderId":"1200468845536492288","transactTime":"1657843038204","price":"1","origQty":"10","executedQty":"0","status":"CANCELED","timeInForce":"GTC","type":"LIMIT","side":"SELL"}}
```

The request above was to stop this obscurity.

Reply:
I understood your point. I have made the above changes.

Reply:
`1200472911738059520 order already cancelled` 👍